File: scripts/lib/atproto/clients/feed.py
from __future__ import annotations

from typing import Optional

from ..transport import TransportLayer


class FeedClient:
    def __init__(self, transport: TransportLayer):
        self._t = transport

    def get_profile(self, actor: str, token: Optional[str] = None) -> dict:
        return self._t.get("app.bsky.actor.getProfile", {"actor": actor}, token=token)

    def get_timeline(self, token: str, limit: int = 50) -> dict:
        return self._t.get("app.bsky.feed.getTimeline", {"limit": limit}, token=token)

    def get_author_feed(self, actor: str, token: Optional[str] = None, limit: int = 50) -> dict:
        return self._t.get(
            "app.bsky.feed.getAuthorFeed", {"actor": actor, "limit": limit}, token=token
        )

    def get_post_thread(self, uri: str, token: Optional[str] = None) -> dict:
        return self._t.get("app.bsky.feed.getPostThread", {"uri": uri}, token=token)

    def get_likes(self, uri: str, token: Optional[str] = None, limit: int = 50) -> dict:
        return self._t.get("app.bsky.feed.getLikes", {"uri": uri, "limit": limit}, token=token)

    def search_actors(self, query: str, token: Optional[str] = None, limit: int = 10) -> dict:
        return self._t.get(
            "app.bsky.actor.searchActors", {"q": query, "limit": limit}, token=token
        )

    def get_actor_likes(
        self, actor: str, token: Optional[str] = None, limit: int = 50
    ) -> dict:
        return self._t.get(
            "app.bsky.feed.getActorLikes", {"actor": actor, "limit": limit}, token=token
        )

    def get_posts(self, uris: list[str], token: Optional[str] = None) -> dict:
        return self._t.get(
            "app.bsky.feed.getPosts", {"uris": ",".join(uris)}, token=token
        )

    def get_reposted_by(
        self, uri: str, token: Optional[str] = None, limit: int = 50
    ) -> dict:
        return self._t.get(
            "app.bsky.feed.getRepostedBy", {"uri": uri, "limit": limit}, token=token
        )

    def get_feed(self, feed_uri: str, token: str, limit: int = 50) -> dict:
        return self._t.get(
            "app.bsky.feed.getFeed", {"feed": feed_uri, "limit": limit}, token=token
        )

    def get_feed_generators(self, uris: list[str], token: Optional[str] = None) -> dict:
        return self._t.get(
            "app.bsky.feed.getFeedGenerators", {"uris": ",".join(uris)}, token=token
        )

File: scripts/lib/deno/clients/feed.ts
import { TransportLayer } from "../transport.ts";

export class FeedClient {
  constructor(private transport: TransportLayer) {}

  async getProfile(actor: string, token?: string) {
    return await this.transport.get("app.bsky.actor.getProfile", { actor }, token);
  }

  async getTimeline(token: string, limit = 50) {
    return await this.transport.get("app.bsky.feed.getTimeline", { limit }, token);
  }

  async getAuthorFeed(actor: string, options: { token?: string; limit?: number } = {}) {
    return await this.transport.get(
      "app.bsky.feed.getAuthorFeed",
      { actor, limit: options.limit ?? 50 },
      options.token
    );
  }

  async getPostThread(uri: string, token?: string) {
    return await this.transport.get("app.bsky.feed.getPostThread", { uri }, token);
  }

  async getLikes(uri: string, options: { token?: string; limit?: number } = {}) {
    return await this.transport.get(
      "app.bsky.feed.getLikes",
      { uri, limit: options.limit ?? 50 },
      options.token
    );
  }

  async searchActors(query: string, options: { token?: string; limit?: number } = {}) {
    return await this.transport.get(
      "app.bsky.actor.searchActors",
      { q: query, limit: options.limit ?? 10 },
      options.token
    );
  }

  async getActorLikes(actor: string, options: { token?: string; limit?: number } = {}) {
    return await this.transport.get(
      "app.bsky.feed.getActorLikes",
      { actor, limit: options.limit ?? 50 },
      options.token
    );
  }

  async getPosts(uris: string[], token?: string) {
    return await this.transport.get(
      "app.bsky.feed.getPosts",
      { uris: uris.join(",") },
      token
    );
  }

  async getRepostedBy(uri: string, options: { token?: string; limit?: number } = {}) {
    return await this.transport.get(
      "app.bsky.feed.getRepostedBy",
      { uri, limit: options.limit ?? 50 },
      options.token
    );
  }

  async getFeed(feedUri: string, token: string, limit = 50) {
    return await this.transport.get(
      "app.bsky.feed.getFeed",
      { feed: feedUri, limit },
      token
    );
  }

  async getFeedGenerators(uris: string[], token?: string) {
    return await this.transport.get(
      "app.bsky.feed.getFeedGenerators",
      { uris: uris.join(",") },
      token
    );
  }
}
